src/logging.cc

```diff
@@ -402,6 +402,13 @@ static bool SendEmailInternal(const char*dest, const char *subject,
 base::Logger::~Logger() {
 }
 
+#ifdef GLOG_ENABLE_CUSTOM_PREFIX
+namespace  {
+  // Optional user-configured callback to print custom prefixes.
+  static PrefixAttacherFunction prefix_attacher_callback = NULL;
```

Comment:
Does it make sense to define a callback per thread? If yes, TLS must be used.

Reply:
I'm not sure. Could we implement that as a separate PR if necessary?

Reply:
On second though, this is probably unnecessary anyway.